stack/services/src/main/java/org/apache/usergrid/security/shiro/utils/SubjectUtils.java

```diff
@@ -243,14 +243,18 @@ public static ApplicationInfo getApplication( Identifier identifier ) {
         String applicationName = null;
         UUID applicationId = null;
         BiMap<UUID, String> applications = getApplications();
+        Set<String> values= applications.values();
+
         if ( applications == null ) {
             return null;
         }
         if ( identifier.isName() ) {
             applicationName = identifier.getName().toLowerCase();
-            applicationId = applications.inverse().get( applicationName );
-            if ( applicationId == null ) {
-                applicationId = applications.inverse().get( identifier.getName() );
+            for (String value: values ){
```

Comment:
The reason I normalize the string is just defensive, the real reason I need to normalize the string is because the value that gets created in the Bimap from getApplications is mixed case and it get instantiated through a Session object. Thats why I take this approach.